drivers/flash/flash_rpc/flash_rpc_host.c

```diff
@@ -0,0 +1,152 @@
+
+#include <nrf_rpc_cbor.h>
+#include <nrf_rpc/nrf_rpc_ipc.h>
+
+#include <zcbor_common.h>
+#include <zcbor_decode.h>
+#include <zcbor_encode.h>
+
+
+LOG_MODULE_DECLARE(FLASH_RPC, CONFIG_FLASH_RPC_LOG_LEVEL);
+
+#define CBOR_BUF_FLASH_MSG_SIZE (sizeof(void *) + sizeof(size_t) + sizeof(off_t))
+
+NRF_RPC_IPC_TRANSPORT(flash_rpc_api_tr, DEVICE_DT_GET(DT_NODELABEL(ipc0)), "flash_rpc_api_ept");
+NRF_RPC_GROUP_DEFINE(flash_rpc_api, "flash_rpc_api", &flash_rpc_api_tr, NULL, NULL, NULL);
+
+static const struct device *const flash_controller =
+	DEVICE_DT_GET_OR_NULL(DT_CHOSEN(zephyr_flash_controller));
+
+static bool decode_flash_msg(struct nrf_rpc_cbor_ctx *ctx, off_t *offset, void **ptr, size_t *len)
+{
+	uint32_t ptr_addr;
+
+	if (!zcbor_uint32_decode(ctx->zs, (uint32_t *)offset)) {
+		return false;
+	}
+	if (zcbor_uint32_decode(ctx->zs, &ptr_addr)) {
+		*ptr = (void *)ptr_addr;
+	} else {
+		return false;
+	}
+
+	return zcbor_uint32_decode(ctx->zs, (uint32_t *)len);
+}
+
+static void rsp_error_code_send(const struct nrf_rpc_group *group, int err_code)
+{
+	struct nrf_rpc_cbor_ctx ctx;
+
+	NRF_RPC_CBOR_ALLOC(group, ctx, sizeof(int));
+
+	zcbor_int32_put(ctx.zs, err_code);
+
+	nrf_rpc_cbor_rsp_no_err(group, &ctx);
+}
+
+static void flash_rpc_init_handler(const struct nrf_rpc_group *group,
+					   struct nrf_rpc_cbor_ctx *ctx, void *handler_data)
+{
+	ARG_UNUSED(handler_data);
+	nrf_rpc_cbor_decoding_done(group, ctx);
+
+	if (!device_is_ready(flash_controller)) {
+		LOG_ERR("Flash Init failure");
+		rsp_error_code_send(group, -EINVAL);
+		return;
+	}
+
+	rsp_error_code_send(group, 0);
+}
+
+static void flash_read_handler(const struct nrf_rpc_group *group, struct nrf_rpc_cbor_ctx *ctx,
+		       void *handler_data)
+{
+	int err;
+	size_t len;
+	off_t offset;
+	void *buffer = NULL;
+
+	if (!decode_flash_msg(ctx, &offset, &buffer, &len)) {
+		err = -EBADMSG;
+		goto error_exit;
+	}
+
+	LOG_DBG("buffer_ptr: %p, offset: 0x%x, len: %d", buffer, (uint32_t)offset, (uint32_t)len);
+	err = flash_read(flash_controller, offset, buffer, len);
+
+error_exit:
+	nrf_rpc_cbor_decoding_done(group, ctx);
+	rsp_error_code_send(group, err);
+}
+
+static void flash_write_handler(const struct nrf_rpc_group *group, struct nrf_rpc_cbor_ctx *ctx,
+			void *handler_data)
+{
+	int err;
+	off_t offset;
+	size_t len;
+	const void *data = NULL;
+
+	if (!decode_flash_msg(ctx, &offset, (void *)&data, &len)) {
+		err = -EBADMSG;
+		goto error_exit;
+	}
+
+	LOG_DBG("offset: %x, data_ptr: %p, len: %d", (uint32_t)offset, data, (uint32_t)len);
+	err = flash_write(flash_controller, offset, data, len);
+
+error_exit:
+	nrf_rpc_cbor_decoding_done(group, ctx);
+	rsp_error_code_send(group, err);
+}
+
+static void flash_erase_handler(const struct nrf_rpc_group *group, struct nrf_rpc_cbor_ctx *ctx,
+		       void *handler_data)
+{
+	int err;
+	size_t size;
+	off_t offset;
+	void *buffer = NULL;
+
+	if (!decode_flash_msg(ctx, &offset, &buffer, &size)) {
+		LOG_INF("Decode failure");
```

Comment:
Other functions do not do that? And shouldn't this be LOG_ERR?

Reply:
moved it and changed it